### DISCUSSION TITLE
Fix for Cura Preprocessing issue

### DISCUSSION THREAD

**Discussion by frank-bordonaro** [FormerLurker/Octolapse#892] [General]
Hey there.

I was having a heck of a time getting [this](https://github.com/FormerLurker/Octolapse/wiki/V0.4---Automatic-Slicer-Configuration#if-you-are-using-cura-follow-these-steps) to work with my combination of Octolapse and Cura (5.3.0) and recently got it to work, so I thought I would share.

My setup is simple:
- Lulzbot Taz 6
- Octopi 1.0
- Octoprint 1.8.7
- Octolapse 0.4.2

The problem seemed straightforward, Auto Detection wasn't working because Octolapse couldn't find the GCode it needed. I looked, and it was there and the variables had actually filled in on the exported gcode, but I noticed a few of the variables it specifically said it needed hadn't populated. My assumption was that I had them either typed wrong, or they were now formatted differently.

![Untitled 2](https://user-images.githubusercontent.com/30547539/230651898-832fc6a1-7ca7-46e0-be94-f30028c047e1.png)

The FIX:
Unrelated, I wanted to add more metadata to my OctoPrint console. I came across [this plugin](https://plugins.octoprint.org/plugins/extrafileinfo/) and saw that it had a very similar need for Cura to send additional settings via gcode. 

![Untitled](https://user-images.githubusercontent.com/30547539/230651203-c78b0a7a-a167-455d-8a2d-b662812f6c49.png)

If you use Cura, you need to add [this](https://gist.github.com/tjjfvi/75210b2ed20ed194d6eab48bf70c4f12) to your start or end gcode. More gcode variables that Cura supports can be found [on this page](http://files.fieldofview.com/cura/Replacement_Patterns.html).

It is probably way overkill to add all of those, but I went ahead and adde them to my end gcode and now Octolapse can get what it needs via Autodetection.

I hope this helps someone else. :)

**Comment by FormerLurker** [comment]
I'm very glad you got it to work!  However, where did you get the script pictured above?  The guide you linked to has [this script](https://raw.githubusercontent.com/wiki/FormerLurker/Octolapse/version/0.4/assets/scripts/slicer/cura/4.2/settings_output.gcode) linked for v4.2+, so not sure what happened there.  I also checked the 4.1 and below script, and the multi extruder scripts, and couldn't find the script you show above.  If it is floating around somewhere, i'd like to find it and fix it.

Edit:  fix broken link above.

**Reply by frank-bordonaro** [reply]
I originally started with the 4.2+ script and when the auto detect wasn't working for me, I tried custom settings. Once I switched to custom, the error was a little more specific as to which parameters it was looking for so I tried adding them one by one without much success.

**Reply by FormerLurker** [reply]
Gotcha.  Any suggestions for how to improve the documentation or in-app help to make all this more clear?

**Reply by frank-bordonaro** [reply]
Honestly, once I figured out that I was just missing some specific parameters, I just used the 'shotgun' approach and added them all. Maybe a suggestion would be to link the [full list of Cura settings](https://gist.github.com/tjjfvi/75210b2ed20ed194d6eab48bf70c4f12) in case people like myself have issues with the existing tutorial since I may be one of a few. It looks like it's by the same author as the Cura 4.2+ code you have, but the complete list of all settings.

**Reply by FormerLurker** [reply]
Yeah, i trimmed the unnecessary commands out.  There actually is a link to all settings in the script itself (look at yours at the very top, it links to https://csi.t6.fyi/, with all the parameters).

Can you do me a favor and try (after making a backup) replacing the current settings script with [this one](https://raw.githubusercontent.com/wiki/FormerLurker/Octolapse/version/0.4/assets/scripts/slicer/cura/4.2/settings_output.gcode)?  I just want to verify that it is working properly for v5.3.  Maybe there are some differences I missed?

**Reply by frank-bordonaro** [reply]
Sure, give me a couple of days to get a break the action and I'll give it a try.
